### DISCUSSION TITLE
Transparent proxy on Fedora - doesn't seem to work

### DISCUSSION THREAD

**Discussion by marcinjahn** [mitmproxy/mitmproxy#5887] [Q&A] [answered]
Hello, I am running Fedora 37. I am trying to setup mitmproxy in the transparent mode, since it seems to be the best solution to have all the traffic go through mitmproxy (as I understand, WireGuard might not work for me, since my mitmproxy server and client machine are the same device).

I followed the steps at https://docs.mitmproxy.org/stable/howto-transparent/.

Basically, I run these commands:

```sh
sysctl -w net.ipv4.ip_forward=1
sysctl -w net.ipv6.conf.all.forwarding=1

sysctl -w net.ipv4.conf.all.send_redirects=0

iptables -t nat -A PREROUTING -i tun0 -p tcp --dport 80 -j REDIRECT --to-port 8080
iptables -t nat -A PREROUTING -i tun0 -p tcp --dport 443 -j REDIRECT --to-port 8080
ip6tables -t nat -A PREROUTING -i tun0 -p tcp --dport 80 -j REDIRECT --to-port 8080
ip6tables -t nat -A PREROUTING -i tun0 -p tcp --dport 443 -j REDIRECT --to-port 8080

mitmproxy --mode transparent --showhost
```

The only thing that I changed compared to the docs is the interface in `iptables` command - I used `tun0` instead of `eth0`, because I am connected to VPN via openvpn, and I need that connection since the target server (the one that the traffic to I want to investigate) is only accessible via VPN.

The next thing i did was to open http://mitm.it/ to install the TLS certificates. Unfortunately, I see the message:

> If you can see this, traffic is not passing through mitmproxy.

I also see that mitmproxy terminal window has en empty list, no matter what website I visit. As I understand, I should be seeing all the traffic there.

Do you know what I am missing?

**Comment by mhils** [comment, marked as answer]
DId you see https://docs.mitmproxy.org/stable/howto-transparent/#work-around-to-redirect-traffic-originating-from-the-machine-itself?

**Comment by marcinjahn** [comment]
I don't know how I could've missed that when it's written so clearly just below the steps that I followed. It obviously works great now, thanks @mhils!
